File: u3u2/Assets/Scripts/app/net/c2s/CGSendMail.cs
using System;
using System.IO;
namespace app.net
{

/**
 * 发送邮件
 * 
 * @author CodeGenerator, don't modify this file please.
 */
public class CGSendMail :BaseMessage
{
	
	/** 收件人名称 */
	private string recName;
	/** 邮件标题 */
	private string title;
	/** 邮件内容 */
	private string content;
	
	public CGSendMail ()
	{
	}
	
	public CGSendMail (
			string recName,
			string title,
			string content )
	{
			this.recName = recName;
			this.title = title;
			this.content = content;
	}
	
	protected override void ReadImpl() 
	{
		return;
	}
	
	protected override void WriteImpl() 
	{
	// 收件人名称
	WriteString(recName);
	// 邮件标题
	WriteString(title);
	// 邮件内容
	WriteString(content);

	}
	
	public override short GetMessageType()
	{
		return (short)MessageType.CG_SEND_MAIL;
	}
	
	public override string getEventType()
	{
		return "";
	}
	
	}
}

File: server/当前版本/server/zlj/game_server/src/com/imop/lj/gameserver/mail/msg/CGSendMail.java
package com.imop.lj.gameserver.mail.msg;

import com.imop.lj.core.msg.MessageType;
import com.imop.lj.gameserver.common.msg.CGMessage;
import com.imop.lj.gameserver.mail.handler.MailHandlerFactory;

/**
 * 发送邮件
 * 
 * @author CodeGenerator, don't modify this file please.
 */
public class CGSendMail extends CGMessage{
	
	/** 收件人名称 */
	private String recName;
	/** 邮件标题 */
	private String title;
	/** 邮件内容 */
	private String content;
	
	public CGSendMail (){
	}
	
	public CGSendMail (
			String recName,
			String title,
			String content ){
			this.recName = recName;
			this.title = title;
			this.content = content;
	}
	
	@Override
	protected boolean readImpl() {

	// 收件人名称
	String _recName = readString();
	//end


	// 邮件标题
	String _title = readString();
	//end


	// 邮件内容
	String _content = readString();
	//end



			this.recName = _recName;
			this.title = _title;
			this.content = _content;
		return true;
	}
	
	@Override
	protected boolean writeImpl() {

	// 收件人名称
	writeString(recName);


	// 邮件标题
	writeString(title);


	// 邮件内容
	writeString(content);


		return true;
	}
	
	@Override
	public short getType() {
		return MessageType.CG_SEND_MAIL;
	}
	
	@Override
	public String getTypeName() {
		return "CG_SEND_MAIL";
	}

	public String getRecName(){
		return recName;
	}
		
	public void setRecName(String recName){
		this.recName = recName;
	}

	public String getTitle(){
		return title;
	}
		
	public void setTitle(String title){
		this.title = title;
	}

	public String getContent(){
		return content;
	}
		
	public void setContent(String content){
		this.content = content;
	}


	@Override
	public void execute() {
		MailHandlerFactory.getHandler().handleSendMail(this.getSession().getPlayer(), this);
	}
}